packages/swapkit/cli/src/screens/ConfigInit.tsx

```diff
@@ -0,0 +1,239 @@
+import SelectInput from '@inkkit/ink-select-input';
+import TextInput from '@inkkit/ink-text-input';
+import type { ConnectConfig } from '@swapkit/sdk';
+import { Box, Text } from 'ink';
+import React, { useCallback, useContext, useEffect, useState } from 'react';
+
+import { NavigationContext } from '../source.js';
+import type { ConfigEditItems } from '../types/navigation.js';
+import { CONFIG_EDIT_ITEMS as CEI } from '../types/navigation.js';
+import { ConfigContext, saveConfig } from '../util/useConfig.js';
+
+const ConfigInit = () => {
+  const [query, setQuery] = useState('');
+  const [goBack, setGoBack] = useState(false);
+  const [updateConfig, setUpdateConfig] = useState(false);
+
+  const { setNavigation } = useContext(NavigationContext);
+  const { config, setConfig } = useContext(ConfigContext);
+
+  const [focus, setFocus] = useState<ConfigEditItems | -1>(-1);
+
+  const [lastSelect, setLastSelect] = useState(0);
+
+  const [inputFocus, setInputFocus] = useState(0);
+  const [query2, setQuery2] = useState('');

```

Comment:
seems like there is overload on `useState`.  I would either recommend going into `useReducer` or directly to sth like `zustand`